addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactBroadcastListener.java

```diff
@@ -0,0 +1,200 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.keba.handler;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.CancelledKeyException;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.ClosedSelectorException;
+import java.nio.channels.DatagramChannel;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link KeContactBroadcastListener} is responsible for receiving UDP broadcast messages sent by the KEBA Charging
+ * Stations. {@link KeContactHandler} willing to receive these messages have to register themselves with the
+ * {@link KeContactBroadcastListener}
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+
+class KeContactBroadcastListener {
+
+    private static final String THING_HANDLER_THREADPOOL_NAME = "thingHandler";
+    public static final int LISTENER_PORT_NUMBER = 7090;
+    public static final int LISTENING_INTERVAL = 100;
+
+    private DatagramChannel broadcastChannel;
+    private SelectionKey broadcastKey;
+    private Selector broadcastListener;
+    private ScheduledFuture<?> broadcastJob;
+    private boolean isStarted = false;
+    protected final ScheduledExecutorService scheduler = ThreadPoolManager
+            .getScheduledPool(THING_HANDLER_THREADPOOL_NAME);
+    private Set<KeContactHandler> listeners = Collections.synchronizedSet(new HashSet<KeContactHandler>());
+
+    private final Logger logger = LoggerFactory.getLogger(KeContactBroadcastListener.class);
+
+    public void start() {
+
+        if (!isStarted) {
+            logger.info("Starting the Keba broadcast listener");
+            try {
+                broadcastChannel = DatagramChannel.open();
+                broadcastChannel.socket().bind(new InetSocketAddress(LISTENER_PORT_NUMBER));
+                broadcastChannel.configureBlocking(false);
+
+                broadcastListener = Selector.open();
+
+                logger.info("Listening for incoming data on {}", broadcastChannel.getLocalAddress());
+
+                synchronized (broadcastListener) {
+                    broadcastListener.wakeup();
+                    broadcastKey = broadcastChannel.register(broadcastListener, broadcastChannel.validOps());
+                }
+
+                if (broadcastJob == null || broadcastJob.isCancelled()) {
+                    broadcastJob = scheduler.scheduleWithFixedDelay(broadcastRunnable, 0, LISTENING_INTERVAL,
```

Comment:
@kaikreuzer It was the Quartz scheduler that went tits-up on frequent calls ;-)

Here it works nicely (It already did before this PR). I just did it with an intervalled scheduling in order not to permanently block a Thread of the ThingHandler thread pool (well, I personally do not like any "wild-running" threads in the JVM, so I prefer to keep them all I the ESH-.... thread pools, in case you were referring to a complete separate Thread)

If I would increase the interval too high then data returned from the Keba will be concatenated, and would complicate the data processing (it is  sending 1 piece of data per UDP packet)